docs/examples/Custom_Pandas_Dataloader.py

```diff
@@ -0,0 +1,280 @@
+#!/usr/bin/env python
+
+"""
+Use of Custom Readers that support Pandas DataFrames
+====================================================
+This is a demonstration of using customised readers that
+support data contained within Pandas DataFrames, rather than
+loading directly from a .csv file using :class:`~.CSVGroundTruthReader` or 
+:class:`~.CSVDetectionReader`. 
+
+The benefit is that this allows us to use the versatile data loading
+capabilities of pandas to read from many different data source types 
+as needed, including .csv, JSON, XML, Parquet, HDF5, .txt, .zip and more. 
+The resulting DataFrame can then simply be fed into the defined 
+`DataFrameGroundTruthReader` or `DataFrameDetectionReader` for further processing
+in Stonesoup as required. 
+"""
+
+# %%
+# Software dependencies
+# ---------------------
+# Before beginning this example, you need to ensure that Pandas is installed, 
+# which is a fast, powerful and flexible open-source data analysis tool in Python.
+# The easiest way to install pandas (if not done so already), is to run pip install
+# from a terminal window within the desired environment:
+#
+# .. code::
+#
+#     pip install pandas
+
+# %%
+# The main dependencies and imports for this example are included below:
+
+import numpy as np
+import os
+import pandas as pd
+
+from stonesoup.base import Property
+from stonesoup.buffered_generator import BufferedGenerator
+from stonesoup.reader.base import GroundTruthReader, DetectionReader, Reader
+from stonesoup.types.groundtruth import GroundTruthPath, GroundTruthState
+
+from typing import Sequence, Collection
+
+from datetime import datetime, timedelta
+from dateutil.parser import parse
+
+# %%
+# Data Frame Reader
+# ^^^^^^^^^^^^^^^^^
+# Similarly to Stonesoups :class:`~._CSVFrameReader`, we'll define a `_DataFrameReader`
```

Comment:
```suggestion
# Similarly to Stone Soup's :class:`~._CSVFrameReader`, we'll define a `_DataFrameReader`
```

Reply:
Morning Oliver, that's no problem at all and totally understood - consistency is important. I'll have a look and ensure this is all consistent for any further changes made.